docs/csharp/pattern-matching.md

```diff
@@ -2,31 +2,225 @@
 title: Pattern Matching | C# Guide
 description: Learn about pattern matching expressions in C#
 keywords: .NET, .NET Core, C#
-ms.date: 10/04/2016
+ms.date: 01/24/2017
+ms.author: wiwagn
 ms.topic: article
 ms.prod: .net
 ms.technology: devlang-csharp
 ms.devlang: csharp
 ms.assetid: 1e575c32-2e2b-4425-9dca-7d118f3ed15b
 ---
 
-# 🔧 Pattern Matching
+# Pattern Matching #
 
-> **Note**
-> 
-> This topic hasn’t been written yet! 
->
-> We welcome your input to help shape the scope and approach. You can track the status and provide input on this
-> [issue](https://github.com/dotnet/docs/issues/1114) at GitHub.
-> 
-> If you would like to review early drafts and outlines of this topic, please leave a note with your contact information in the issue.
->
-> Learn more about how you can contribute on [GitHub](https://github.com/dotnet/docs/blob/master/CONTRIBUTING.md).
->
+Patterns test that a value has a certain *shape*, and can *extract*
+information from the value when it has the matching shape. Pattern
+matching provides more concise syntax for algorithms you already use
+today. You already create pattern matching algorithms using existing
+syntax. You write `if` or `switch` statements that test values. Then,
+when those statements match, you extract and use information from that
+value. The new syntax elements are extensions to statements you are already
+familiar with: `is` and `switch`. These new extensions combine testing
+a value and extracting that information.
 
-<!--
+In this topic, we'll look at the new syntax to show you how it enables
+readable, concise code. Pattern matching enables idioms you'll need more
+often building distributed systems, where data and the methods that manipulate
+that data are separate. 
```

Comment:
An awkward sentence. Perhaps "Pattern matching enables idioms that you use most often when building..."